### PR TITLE
fix windows tests

### PR DESCRIPTION
**Pull request by TheRoyalTnetennba**

### CODE REVIEW

File: google-cloud-bigquery/test/google/cloud/bigquery/dataset_insert_test.rb [hunk 1]
```diff
@@ -240,7 +240,7 @@
...
+      json: {"id"=>2, "name"=>"Gandalf", "age"=>1000, "weight"=>198.6, "is_magic"=>true, "scores"=>[100.0, 99.0, 0.001], "spells"=>[{"name"=>"Skydragon", "discovered_by"=>"Firebreather", "properties"=>[{"name"=>"Flying", "power"=>1.0}, {"name"=>"Creature", "power"=>1.0}, {"name"=>"Explodey", "power"=>11.0}], "icon"=> Google::Cloud::Bigquery::Convert.to_json_value(File.open("acceptance/data/kitten-test-data.json", "rb")), "last_used"=>"2015-10-31 23:59:56.000000+00:00"}], "tea_time"=>"15:00:00", "next_vacation"=>"2666-06-06", "favorite_time"=>"2001-12-19 23:59:59.000000"}
```
Comment (blowmage) [inline google-cloud-bigquery/test/google/cloud/bigquery/dataset_insert_test.rb]: I believe this has passed on Windows previously. Do you know why it is not matching now?

Comment (TheRoyalTnetennba) [reply google-cloud-bigquery/test/google/cloud/bigquery/dataset_insert_test.rb]: I don't. The icon was slightly different from the expected. 🤷‍♂️

Comment (blowmage) [reply google-cloud-bigquery/test/google/cloud/bigquery/dataset_insert_test.rb]: Ruby will read text files on Windows differently than on POSIX, but the `b` in the read mode should suppress this.

Are the files in the `acceptance/data` directory being modified or changed somehow in the current Windows setup?

Comment (TheRoyalTnetennba) [reply google-cloud-bigquery/test/google/cloud/bigquery/dataset_insert_test.rb]: The only place I know of that touches the acceptance data is https://github.com/googleapis/google-cloud-ruby/blob/0a2b7112593e5520db19ab3ba5588fc286f211ef/Rakefile#L632 . The files are identical on the windows machine I'm using to test currently.

File: google-cloud-core/lib/google/cloud.rb [hunk 2]
```diff
@@ -204,7 +204,10 @@ def self.auto_load_gems
     #
     def self.loaded_files
       files = Array(caller).map do |backtrace_line|
-        backtrace_line.split(":").first
+        until backtrace_line.split(":").size == 1 || File.file?(backtrace_line)
```
[the remark this answers is not in the document]
Comment (TheRoyalTnetennba) [reply google-cloud-core/lib/google/cloud.rb]: What would you envision for that?

[the remark this answers is not in the document]
Comment (blowmage) [reply google-cloud-core/lib/google/cloud.rb]: Something that shows why this change is needed. See `google-cloud-core/test/google/cloud/loaded_files_test.rb`.

**Comment by TheRoyalTnetennba** [comment]
> @TheRoyalTnetennba Is this change still needed?

Should be. There was another fix I was hoping to squeeze in, but it obviously didn't happen. Merging as long as it's still greener than master.

**Pull request merged**